### PR TITLE
feat: improve new PR and issue labeling for triaging

### PR DESCRIPTION
**Pull request by torresxb1**
#### Why is this change necessary?
Will help triage which new issues and PRs haven't been looked at by the team, as querying by no comments or no labels can sometimes be misleading.

#### How does it address the issue?
New label `stage/needs-triage` automatically assigned on new issues and PRs.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+blank_issues_enabled: false
```
Comment (torresxb1) [inline .github/ISSUE_TEMPLATE/config.yml:1]: disabled blank issues and added a `Anything else` template, in order to ensure that all new issues are auto-labeled with `needs-triage`

File: .github/ISSUE_TEMPLATE/anything_else.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+name: Anything else
+about: Choose if your issue doesn't apply to the other templates
```
Comment (torresxb1) [inline .github/ISSUE_TEMPLATE/anything_else.md:3]: open to other suggestions

File: .github/ISSUE_TEMPLATE/anything_else.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+name: Anything else
```
Comment (jfuss) [inline .github/ISSUE_TEMPLATE/anything_else.md:2]: I wonder if we should say "Other" than "Anything else". More of a nit and not something I have strong feelings on and something we can always change later.

Comment (torresxb1) [reply .github/ISSUE_TEMPLATE/anything_else.md]: done, no strong preference either (originally debated between the two lol)

**Review by jfuss — APPROVED** [review]

**Review by hawflau — APPROVED** [review]

**Pull request merged**